src/main/kotlin/org/rust/ide/typing/paste/RsConvertJsonToStructCopyPasteProcessor.kt

```diff
@@ -196,22 +196,113 @@ private fun createAndInsertStruct(
     return inserted
 }
 
-fun generateStruct(
+private fun Char.isSeparator(): Boolean = this == '_' || this == '-'
+
+// fooBarBaz
+private fun String.isCamelCase(): Boolean =
+    isNotEmpty()
+        && first().isLowerCase()
+        && !contains("__")
+        && !contains("--")
+        && !zipWithNext().any { (fst, snd) -> (fst.hasCase && snd.isSeparator()) || (snd.hasCase && fst.isSeparator()) }
+
+// FooBarBaz
+private fun String.isPascalCase(): Boolean =
+    isNotEmpty()
+        && first().isUpperCase()
+        && !contains("__")
+        && !contains("--")
+        && !zipWithNext().any { (fst, snd) -> (fst.hasCase && snd.isSeparator()) || (snd.hasCase && fst.isSeparator()) }
+
+// foo_bar_baz
+private fun String.isSnakeCase(): Boolean = isNotEmpty() && none { it.isUpperCase() || it == '-' }
+
+// foo-bar-baz
+private fun String.isKebabCase(): Boolean = isNotEmpty() && none { it.isUpperCase() }
+
+// FOO_BAR_BAZ
+private fun String.isScreamingSnakeCase(): Boolean = isNotEmpty() && none { it.isLowerCase() || it == '-' }
+
+// FOO-BAR-BAZ
+private fun String.isScreamingKebabCase(): Boolean = isNotEmpty() && none { it.isLowerCase() }
+
+private const val DEFAULT_CONVENTION = "snake_case";
+
+/**
+ * Tries to guess the naming convention from the field name.
+ * If the convention matches snake case, no renaming will be applied, since it is the default.
+ */
+private fun getNamingConvention(field: String): String = when {
+    // The order of the checks is important
+    field.isSnakeCase() -> DEFAULT_CONVENTION
+    field.isKebabCase() -> "kebab-case"
+    field.isScreamingSnakeCase() -> "SCREAMING_SNAKE_CASE"
+    field.isScreamingKebabCase() -> "SCREAMING-KEBAB-CASE"
+    field.isPascalCase() -> "PascalCase"
+    field.isCamelCase() -> "camelCase"
+    else -> DEFAULT_CONVENTION
```

Comment:
What about `UPPERCASE`? Though technically it can be emulated as `SCREAMING_SNAKE_CASE`, I think it is worth to have it as separate variant